src/Compilers/Core/AnalyzerDriver/AnalyzerExecutor.cs

```diff
@@ -25,6 +26,8 @@ internal class AnalyzerExecutor
         private readonly Action<Exception, DiagnosticAnalyzer, Diagnostic> _onAnalyzerException;
         private readonly AnalyzerManager _analyzerManager;
         private readonly Func<DiagnosticAnalyzer, bool> _isCompilerAnalyzer;
+        private readonly ImmutableDictionary<DiagnosticAnalyzer, object> _singleThreadedAnalyzerToGateMap;
```

Comment:
Can analyzers call back into `AnalyzerExecutor` indirectly?  Say if I ask a symbol for errors?  If so have we sufficiently mitigated against deadlocks in that case?